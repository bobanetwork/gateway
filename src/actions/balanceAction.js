/*
  Varna - A Privacy-Preserving Marketplace
  Varna uses Fully Homomorphic Encryption to make markets fair.
  Copyright (C) 2021 Enya Inc. Palo Alto, CA

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

import networkService from 'services/networkService'
import { createAction } from './createAction'

export function fetchL1LPBalance(address) {
    return createAction('FETCH/L1LP/BALANCE', () => networkService.L1LPBalance(address))
}

export function fetchL2LPBalance(address) {
    return createAction('FETCH/L2LP/BALANCE', () => networkService.L2LPBalance(address))
}

export function fetchClassicExitCost(address) {
    return createAction('FETCH/CLASSICEXIT/COST', () => networkService.getExitCost(address))
}

export function fetchAltL1DepositFee() {
    return createAction('FETCH/ALTL1DEPOSIT/COST', () => networkService.getAltL1DepositFee())
}

export function fetchL2BalanceETH() {
    return createAction('FETCH/L2ETH/BALANCE', () => networkService.getL2BalanceETH())
}

export function fetchL2BalanceBOBA() {
    return createAction('FETCH/L2BOBA/BALANCE', () => networkService.getL2BalanceBOBA())
}

export function fetchExitFee() {
  return createAction('FETCH/EXITFEE', () => networkService.getExitFeeFromBillingContract())
}
