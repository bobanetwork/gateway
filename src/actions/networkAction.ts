/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

import networkService, { EPoolLayer } from 'services/networkService'
import transactionService from 'services/transaction.service'
import { createAction } from './createAction'
import { BigNumberish } from 'ethers'

export const fetchBalances = () =>
  createAction('BALANCE/GET', () => networkService.getBalances())

export const addTokenList = () =>
  createAction('TOKENLIST/GET', () => networkService.addTokenList())

export const fetchTransactions = () =>
  createAction('TRANSACTION/GETALL', () => transactionService.getTransactions())

export const fetchSevens = () =>
  createAction('SEVENS/GETALL', () => transactionService.getSevens())

export const fetchFastExits = () =>
  createAction('FASTEXITS/GETALL', () => transactionService.getFastExits())

export const exitBOBA = (token: string, value: BigNumberish) =>
  createAction('EXIT/CREATE', () => networkService.exitBOBA(token, value))

//SWAP RELATED
export const depositL1LP = (currency: string, value: BigNumberish) =>
  createAction('DEPOSIT/CREATE', () =>
    networkService.depositL1LP(currency, value)
  )

export const isTeleportationOfAssetSupported = (
  layer: string,
  asset: string,
  destChainId: string
) =>
  createAction('DEPOSIT/TELEPORTATION/TOKEN_SUPPORTED', () =>
    networkService.isTeleportationOfAssetSupported(layer, asset, destChainId)
  )

export const depositWithLightBridge = (
  layer: string,
  currency: string,
  value: BigNumberish,
  destChainId: BigNumberish
) =>
  createAction('DEPOSIT/CREATE', () =>
    networkService.depositWithTeleporter(layer, currency, value, destChainId)
  )

//SWAP RELATED - Depositing into the L2LP triggers the swap-exit
export const depositL2LP = (token: string, value: BigNumberish) =>
  createAction('EXIT/CREATE', () => networkService.depositL2LP(token, value))

//CLASSIC DEPOSIT ETH
export const depositETHL2 = (payload) =>
  createAction('DEPOSIT/CREATE', () => {
    return networkService.depositETHL2(payload)
  })

//DEPOSIT ERC20
export const depositErc20 = (payload) =>
  createAction('DEPOSIT/CREATE', () => networkService.depositErc20(payload))

// FIXME: move this earn related actions.
export const getReward = (
  currencyAddress: string,
  value_Wei_String: BigNumberish,
  L1orL2Pool: EPoolLayer
) =>
  createAction('EARN/HARVEST', () =>
    networkService.getReward(currencyAddress, value_Wei_String, L1orL2Pool)
  )

export const withdrawLiquidity = (
  currencyAddress: string,
  value_Wei_String: string,
  L1orL2Pool: EPoolLayer
) =>
  createAction('EARN/WITHDRAW', () =>
    networkService.withdrawLiquidity(
      currencyAddress,
      value_Wei_String,
      L1orL2Pool
    )
  )

export const approveERC20 = (
  value: BigNumberish,
  currency: string,
  approveContractAddress: string,
  contractABI?
) =>
  createAction('APPROVE/CREATE', () =>
    networkService.approveERC20(
      value,
      currency,
      approveContractAddress,
      contractABI
    )
  )

export const transfer = (
  recipient: string,
  value: BigNumberish,
  currency: string
) =>
  createAction('TRANSFER/CREATE', () =>
    networkService.transfer(recipient, value, currency)
  )

export const fetchLookUpPrice = (params) =>
  createAction('PRICE/GET', () => networkService.fetchLookUpPrice(params))

export const clearLookupPrice = () => (dispatch) =>
  dispatch({ type: 'LOOKUP/PRICE/CLEAR' })

export const getAllAddresses = () =>
  createAction('GET/ALL/ADDRESS', () => networkService.getAllAddresses())

/********************************/
/******ONE GATEWAY ACTIONS *****/
/********************************/
/**
 * @param payload: {network: ethereum, bnb, networkType
 * @param networkType:  MAINNET, TESTNET
 */
export const setNetwork = (payload) => (dispatch) =>
  dispatch({ type: 'NETWORK/SET', payload })

// to update the active network.
export const setActiveNetwork = () => (dispatch) =>
  dispatch({ type: 'NETWORK/SET/ACTIVE' })

export const setActiveNetworkType = (payload) => (dispatch) =>
  dispatch({ type: 'NETWORK/SET_TYPE/ACTIVE', payload })
