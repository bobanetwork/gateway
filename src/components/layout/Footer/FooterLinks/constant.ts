interface LinkProps {
  label: string
  path: string
  isNav?: boolean
}

export const FOOTERLINKS: Array<LinkProps> = [
  {
    label: 'FAQs',
    path: 'https://docs.boba.network/faq',
  },
  {
    label: 'Dev Tools',
    path: 'https://docs.boba.network/developer',
  },
  {
    label: 'Boba Network Website',
    path: 'https://boba.network',
  },
  {
    label: 'Terms of Use',
    path: 'https://boba.network/terms-of-use/',
  },
]
