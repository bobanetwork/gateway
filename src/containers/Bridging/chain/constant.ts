import { ElementType } from 'react'

import BNBIcon from 'components/icons/chain/L1/BNBIcon'
import EthereumIcon from 'components/icons/chain/L1/EthereumIcon'

import BobaBNBIcon from 'components/icons/chain/L2/BobaBNBIcon'
import BobaIcon from 'components/icons/chain/L2/BobaIcon'
import OptimismIcon from 'components/icons/chain/L2/OptimismIcon'
import ArbitrumIcon from 'components/icons/chain/L2/ArbitrumIcon'

type IconType = {
  L1: ElementType
  L2: ElementType
}

type NetworkIconsType = Record<string, IconType>

export const NETWORK_ICONS: NetworkIconsType = {
  ethereum: {
    L1: EthereumIcon,
    L2: BobaIcon,
  },
  bnb: {
    L1: BNBIcon,
    L2: BobaBNBIcon,
  },
  optimism: {
    L1: EthereumIcon,
    L2: OptimismIcon,
  },
  arbitrum: {
    L1: EthereumIcon,
    L2: ArbitrumIcon,
  },
}
