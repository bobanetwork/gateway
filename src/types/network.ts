export type NetworkSwitchState =
  | 'idle'
  | 'confirming'
  | 'switching'
  | 'success'
  | 'failed'
  | 'chain-not-found'

export interface NetworkSwitchError {
  code?: number;
  message: string;
}
