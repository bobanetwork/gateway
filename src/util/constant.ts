/**************
 * Env Params *
 **************/
/* eslint-disable @typescript-eslint/no-var-requires */
require('dotenv').config()

export type EnvType = string | number | null | undefined

export const POLL_INTERVAL: EnvType =
  process.env.REACT_APP_POLL_INTERVAL || 20000
export const GAS_POLL_INTERVAL: EnvType =
  process.env.REACT_APP_GAS_POLL_INTERVAL || 40000
export const GA4_MEASUREMENT_ID: EnvType =
  process.env.REACT_APP_GA4_MEASUREMENT_ID || null
export const APP_ENV: EnvType = process.env.REACT_APP_ENV || 'dev'
export const WALLET_VERSION: EnvType = process.env.REACT_APP_WALLET_VERSION
export const WC_PROJECT_ID: EnvType = process.env.REACT_APP_WC_PROJECT_ID
// WalletConnect FLAG
export const DISABLE_WALLETCONNECT: EnvType =
  process.env.REACT_APP_DISABLE_WALLETCONNECT

const GATEWAY_DATA_BRANCH: EnvType =
  process.env.REACT_APP_GATEWAY_DATA_BRANCH || 'main'

/*********************
 * Routes Constants **
 *********************/

type RoutesPathType = {
  BRIDGE: string
  HISTORY: string
  EARN: string
  STAKE: string
  DAO: string
  ECOSYSTEM: string
  TRADE: string
}

export const ROUTES_PATH: RoutesPathType = {
  BRIDGE: '/bridge',
  HISTORY: '/history',
  EARN: '/earn',
  STAKE: '/stake',
  DAO: '/dao',
  ECOSYSTEM: '/ecosystem',
  TRADE: '/trade',
}

type Network = 'ethereum' | 'bnb' | 'optimism' | 'arbitrum'
type Page =
  | 'Bridge'
  | 'History'
  | 'Earn'
  | 'Stake'
  | 'DAO'
  | 'trade'
  | 'ecosystem'
type PagesByNetworkType = Record<Network, Page[]>

export const PAGES_BY_NETWORK: PagesByNetworkType = {
  ethereum: ['Bridge', 'History', 'Earn', 'Stake', 'DAO', 'ecosystem', 'trade'],
  bnb: ['Bridge', 'Earn', 'History', 'ecosystem', 'trade'],
  optimism: ['Bridge', 'History', 'ecosystem', 'trade'],
  arbitrum: ['Bridge', 'History', 'ecosystem', 'trade'],
}

export enum Layer {
  L1 = 'L1',
  L2 = 'L2',
}

export const LAYER: { [key: string]: Layer } = Layer

type NetworkIconType = 'ethereum' | 'boba'

type NetworkNameType = {
  L1: string
  L2: string
}

type DefaultNetworkType = {
  NAME: NetworkNameType
  ICON: Record<Layer, NetworkIconType>
}

export const DEFAULT_NETWORK: DefaultNetworkType = {
  NAME: {
    L1: 'Ethereum',
    L2: 'Boba ETH',
  },
  ICON: {
    L1: 'ethereum',
    L2: 'boba',
  },
}

export const MM_EXTENTION_URL: string =
  'https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn?hl=en'

export const MIN_NATIVE_L1_BALANCE: number = 0.002

export const THIRD_PARTY_BRIDGES_LIST = `https://raw.githubusercontent.com/bobanetwork/gateway-data/${GATEWAY_DATA_BRANCH}/bridges/list.json`

export const COIN_GECKO_URL = `https://api.coingecko.com/api/v3/`

export const ERROR_CODE = 'GATEWAY ERROR:'
