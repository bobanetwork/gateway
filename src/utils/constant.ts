export type EnvType = string | number | null | undefined

export const GATEWAY_DATA_BRANCH: EnvType =
  import.meta.env.VITE_GATEWAY_DATA_BRANCH || 'main'

export const GATEWAY_DATA_RAW =
  'https://raw.githubusercontent.com/bobanetwork/gateway-data/'

export const DATA_URL_THIRD_PARTY_BRIDGE = `https://raw.githubusercontent.com/bobanetwork/gateway-data/${GATEWAY_DATA_BRANCH}/bridges/list.json`

//@todo update branch name
export const DATA_URL_ECOSYSTEM_LIST = `${GATEWAY_DATA_RAW}${GATEWAY_DATA_BRANCH}/ecosystem/list.json`
export const DATA_URL_TRADE_LIST = `${GATEWAY_DATA_RAW}${GATEWAY_DATA_BRANCH}/ecosystem/cex.list.json`
export const DATA_URL_TRADE_ASSET = `${GATEWAY_DATA_RAW}${GATEWAY_DATA_BRANCH}/ecosystem/icons/`
