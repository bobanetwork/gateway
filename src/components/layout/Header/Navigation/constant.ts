import { ROUTES_PATH } from 'util/constant'
import { IMenuItem } from './types'

export const MENU_LIST: Array<IMenuItem> = [
  {
    label: 'Bridge',
    path: ROUTES_PATH.BRIDGE,
  },
  {
    label: 'History',
    path: ROUTES_PATH.HISTORY,
  },
  {
    label: 'Stake',
    path: ROUTES_PATH.STAKE,
  },
  {
    label: 'Dao',
    path: ROUTES_PATH.DAO,
  },
  // {
  //   label: 'Trade',
  //   path: ROUTES_PATH.TRADE,
  // },
  {
    label: 'Ecosystem',
    path: ROUTES_PATH.ECOSYSTEM,
  },
]
