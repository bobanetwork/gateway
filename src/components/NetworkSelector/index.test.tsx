import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import CustomThemeProvider from 'themes'
import { mockedInitialState } from 'util/tests'
import { NetworkSelector } from '.'
import { setNetwork } from 'actions/networkAction'
import { Network, NetworkType } from 'util/network/network.util'
import { Layer } from 'util/constant'

const mockDispatch = jest.fn()
jest.mock('react-redux', () => {
  const originalModule = jest.requireActual('react-redux')
  return {
    __esModule: true,
    ...originalModule,
    useDispatch: () => mockDispatch,
  }
})

jest.mock('actions/networkAction', () => {
  const originalModule = jest.requireActual('actions/networkAction')
  return {
    __esModule: true,
    ...originalModule,
    setNetwork: jest.fn(),
  }
})

const mockStore = configureStore([thunk])

const renderNetworkSelector = ({ store }: any) => {
  return render(
    <Provider store={store}>
      <CustomThemeProvider>
        <NetworkSelector />
      </CustomThemeProvider>
    </Provider>
  )
}

// render the network selector
// ensure that it matches the snapshot
// ensure that when clicked it triggers the setNetwork
// switch to test night by changing the store, ensure that it matches the snapshot and has correct options available
// switch bridge to light, ensure that it matches the snapshot
// ensure that when clicked it triggers the set network

describe('NetworkSelector', () => {
  let store
  beforeEach(() => {
    store = {
      ...mockedInitialState,
      setup: {
        ...mockedInitialState.bridge,
        accountEnabled: true,
        netLayer: 'L1',
      },
      bridge: {
        ...mockedInitialState.bridge,
        bridgeToAddressState: true,
      },
    }
  })

  test('should match snapshot when account connected to Ethereum Mainnet', async () => {
    store = mockStore({
      ...store,
    })

    const { asFragment, getByText } = renderNetworkSelector({
      store,
    })
    expect(getByText('Ethereum')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
    const header = getByText('Ethereum')
    await fireEvent.click(header)
    const binanceOption = getByText('Binance Smart Chain')
    expect(binanceOption).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
    await fireEvent.click(binanceOption)
    expect(setNetwork).toHaveBeenCalledWith({
      network: Network.BNB,
      name: 'Bnb',
      networkIcon: 'bnb',
      chainIds: undefined,
      networkType: NetworkType.MAINNET,
    })
  })

  test('should match snapshot when account connected to Boba (Sepolia)', async () => {
    store = mockStore({
      ...store,
      bridge: {
        ...store.bridge,
        bridgeType: 'LIGHT',
      },
      network: {
        ...store.network,
        activeNetworkName: {
          l1: 'ethereum',
          l2: 'boba',
        },
        activeNetwork: 'ETHEREUM',
        activeNetworkType: 'Testnet',
        activeNetworkIcon: 'ethereum',
        chainIds: { L1: '11155111', L2: '11155420' },
      },
      setup: {
        netLayer: 'L2',
      },
    })

    const { asFragment, getByText } = renderNetworkSelector({
      store,
    })
    expect(getByText('Boba (Sepolia)')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
    const header = getByText('Boba (Sepolia)')
    await fireEvent.click(header)
    const binanceOption = getByText('Optimism (Sepolia)')
    expect(binanceOption).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
    await fireEvent.click(binanceOption)
    expect(setNetwork).toHaveBeenCalledWith({
      network: Network.OPTIMISM,
      name: 'Optimism Sepolia',
      networkIcon: 'optimism',
      chainIds: undefined,
      networkType: NetworkType.TESTNET,
    })
  })
})
