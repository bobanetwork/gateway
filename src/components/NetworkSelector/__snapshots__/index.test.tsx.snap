// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkSelector should match snapshot when account connected to Boba (Sepolia) 1`] = `
<DocumentFragment>
  <div
    class="sc-gFqAkR kUbaAB dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    data-testid="dropdown-network-selector"
  >
    <div
      class="sc-ikkxIA bvEfJd dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    >
      <div
        class="sc-fUnMCh jySLNu"
      >
        <div
          class="sc-dAbbOL ACNsH"
        >
          <img
            alt="token"
            src="bobaEth.svg"
            width="20px"
          />
        </div>
        <p
          class="sc-dhKdcB jSXfaR"
        >
          Boba (Sepolia)
        </p>
        <div
          class="sc-jsJBEP hOLTkK dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div>
            <div
              data-src="arrowdown.svg"
              src=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NetworkSelector should match snapshot when account connected to Boba (Sepolia) 2`] = `
<DocumentFragment>
  <div
    class="sc-gFqAkR kUbaAB dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    data-testid="dropdown-network-selector"
  >
    <div
      class="sc-ikkxIA fydrhM dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    >
      <div
        class="sc-fUnMCh jySLNu"
      >
        <div
          class="sc-dAbbOL ACNsH"
        >
          <img
            alt="token"
            src="bobaEth.svg"
            width="20px"
          />
        </div>
        <p
          class="sc-dhKdcB jSXfaR"
        >
          Boba (Sepolia)
        </p>
        <div
          class="sc-jsJBEP hOLTkK dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div>
            <div
              data-src="arrowdown.svg"
              src=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-hmdomO iBZDnJ"
    >
      <div
        class="sc-bXCLTC cnZnBu"
      >
        <div
          class="sc-feUZmu cWlxKB dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          Networks
        </div>
        <div
          class="sc-fUnMCh gRWBIx dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div
            class="sc-dAbbOL ACNsH"
          >
            <img
              alt="token"
              src="bobaEth.svg"
              width="20px"
            />
          </div>
          Boba (Sepolia)
        </div>
        <div
          class="sc-fUnMCh jySLNu dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div
            class="sc-dAbbOL ACNsH"
          >
            <img
              alt="token"
              src="bobaBNB.svg"
              width="20px"
            />
          </div>
          Boba BNB Testnet
        </div>
        <div
          class="sc-fUnMCh jySLNu dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div
            class="sc-dAbbOL ACNsH"
          >
            <img
              alt="token"
              src="optimism.svg"
              width="20px"
            />
          </div>
          Optimism (Sepolia)
        </div>
        <div
          class="sc-fUnMCh jySLNu dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div
            class="sc-dAbbOL ACNsH"
          >
            <img
              alt="token"
              src="arbitrum.svg"
              width="20px"
            />
          </div>
          Arbitrum (Sepolia)
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NetworkSelector should match snapshot when account connected to Ethereum Mainnet 1`] = `
<DocumentFragment>
  <div
    class="sc-gFqAkR kUbaAB dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    data-testid="dropdown-network-selector"
  >
    <div
      class="sc-ikkxIA bvEfJd dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    >
      <div
        class="sc-fUnMCh jySLNu"
      >
        <div
          class="sc-dAbbOL ACNsH"
        >
          <img
            alt="token"
            src="ethereum.svg"
            width="20px"
          />
        </div>
        <p
          class="sc-dhKdcB jSXfaR"
        >
          Ethereum
        </p>
        <div
          class="sc-jsJBEP hOLTkK dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div>
            <div
              data-src="arrowdown.svg"
              src=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NetworkSelector should match snapshot when account connected to Ethereum Mainnet 2`] = `
<DocumentFragment>
  <div
    class="sc-gFqAkR kUbaAB dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    data-testid="dropdown-network-selector"
  >
    <div
      class="sc-ikkxIA fydrhM dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
    >
      <div
        class="sc-fUnMCh jySLNu"
      >
        <div
          class="sc-dAbbOL ACNsH"
        >
          <img
            alt="token"
            src="ethereum.svg"
            width="20px"
          />
        </div>
        <p
          class="sc-dhKdcB jSXfaR"
        >
          Ethereum
        </p>
        <div
          class="sc-jsJBEP hOLTkK dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div>
            <div
              data-src="arrowdown.svg"
              src=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-hmdomO iBZDnJ"
    >
      <div
        class="sc-bXCLTC cnZnBu"
      >
        <div
          class="sc-feUZmu cWlxKB dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          Networks
        </div>
        <div
          class="sc-fUnMCh gRWBIx dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div
            class="sc-dAbbOL ACNsH"
          >
            <img
              alt="token"
              src="ethereum.svg"
              width="20px"
            />
          </div>
          Ethereum
        </div>
        <div
          class="sc-fUnMCh jySLNu dropdown sc-dtInlm sc-kOPcWz iSOFgg kDoMup"
        >
          <div
            class="sc-dAbbOL ACNsH"
          >
            <img
              alt="token"
              src="https://raw.githubusercontent.com/bobanetwork/token-list/main/assets/bnb.svg"
              width="20px"
            />
          </div>
          Binance Smart Chain
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
